test/presenters/notices/setup/abstraction-alerts/alert-thresholds.presenter.test.js

```diff
@@ -64,6 +68,134 @@ describe('Notices Setup - Abstraction Alerts - Alert Thresholds Presenter', () =
+              },
+              text: '100 m',
+              value: '2'
+            }
+          ])
+        })
+      })
+
+      describe('and the "alertType" is "stop" ', () => {
+        beforeEach(() => {
+          session = {
+            ...AbstractionAlertSessionData.monitoringStation(),
+            alertType: 'stop'
+          }
+        })
+
+        it('returns page data for the view, with only the thresholds with stop restrictions', () => {
+          const result = AlertThresholdsPresenter.go(session)
+
+          expect(result.thresholdOptions).to.equal([
+            {
+              checked: false,
+              hint: {
+                text: 'Flow thresholds for this station (m3/s)'
+              },
+              text: '100 m3/s',
+              value: '1'
+            },
+            {
+              checked: false,
+              hint: {
+                text: 'Level thresholds for this station (m)'
+              },
+              text: '100 m',
+              value: '2'
+            }
+          ])
+        })
+      })
+
+      describe('and the "alertType" is "reduce" ', () => {
+        beforeEach(() => {
+          session = {
+            ...AbstractionAlertSessionData.monitoringStation(),
+            alertType: 'reduce'
+          }
+        })
+
+        it('returns page data for the view, with only the thresholds with stop restrictions', () => {
+          const result = AlertThresholdsPresenter.go(session)
+
+          expect(result.thresholdOptions).to.equal([
+            {
+              checked: false,
+              hint: {
+                text: 'Flow thresholds for this station (m)'
+              },
+              text: '1000 m',
+              value: '0'
+            }
+          ])
+        })
+      })
+
+      describe('and the "alertType" is "warning" ', () => {
+        beforeEach(() => {
+          session = {
+            ...AbstractionAlertSessionData.monitoringStation(),
+            alertType: 'warning'
+          }
+        })
+
+        it('returns page data for the view, with all the thresholds', () => {
+          const result = AlertThresholdsPresenter.go(session)
+
+          expect(result.thresholdOptions).to.equal([
+            {
+              checked: false,
+              hint: {
+                text: 'Flow thresholds for this station (m)'
+              },
+              text: '1000 m',
+              value: '0'
+            },
+            {
+              checked: false,
+              hint: {
+                text: 'Flow thresholds for this station (m3/s)'
+              },
+              text: '100 m3/s',
+              value: '1'
+            },
+            {
+              checked: false,
+              hint: {
+                text: 'Level thresholds for this station (m)'
+              },
+              text: '100 m',
+              value: '2'
+            }
+          ])
+        })
+      })
+
+      describe('and the "alertType" is "resume" ', () => {
```

Comment:
TL;DR; happy for "warning" and "resume" to be combined into "and the alert type is not stop or reduce".

I'm fine in cases like this where we do something if `x`, else do if not `x` do something else to say that in the test description and just use have one test that is not `x`.

For example, if there were 30 different threshold types, and we just do something special for a couple of them, I wouldn't expect to see tests for the other 28.